src/python/grpcio/grpc/__init__.py

```diff
@@ -900,88 +919,78 @@ def add_generic_rpc_handlers(self, generic_rpc_handlers):
-        contained in the passed address was zero.
+      integer:
+      An integer port on which server will accept RPC requests.
     """
         raise NotImplementedError()
 
     @abc.abstractmethod
     def add_secure_port(self, address, server_credentials):
-        """Reserves a port for secure RPC service after this Server becomes active.
+        """Opens a secure port for accepting RPCs.
 
-    This method may only be called before calling this Server's start method is
-    called.
+    This method may only be called before starting the server.
 
     Args:
       address: The address for which to open a port.
-      server_credentials: A ServerCredentials.
+        if the port is 0, or not specified in the address, then gRPC runtime
+        will choose a port.
+      server_credentials: A ServerCredentials object.
 
     Returns:
-      An integer port on which RPCs will be serviced after this link has been
-        started. This is typically the same number as the port number contained
-        in the passed address, but will likely be different if the port number
-        contained in the passed address was zero.
+      integer:
+      An integer port on which server will accept RPC requests.
     """
         raise NotImplementedError()
 
     @abc.abstractmethod
     def start(self):
-        """Starts this Server's service of RPCs.
+        """Starts this Server.
 
-    This method may only be called while the server is not serving RPCs (i.e. it
-    is not idempotent).
+    This method may only be called once. (i.e. it is not idempotent).
     """
         raise NotImplementedError()
 
     @abc.abstractmethod
     def stop(self, grace):
-        """Stops this Server's service of RPCs.
+        """Stops this Server.
 
-    All calls to this method immediately stop service of new RPCs. When existing
-    RPCs are aborted is controlled by the grace period parameter passed to this
-    method.
+    This method immediately stop service of new RPCs in all cases.
+    If grace period is specified, this method returns immediately
+    and all RPCs active at the end of the grace period are aborted.
 
-    This method may be called at any time and is idempotent. Passing a smaller
-    grace value than has been passed in a previous call will have the effect of
-    stopping the Server sooner. Passing a larger grace value than has been
-    passed in a previous call will not have the effect of stopping the server
-    later.
+    If grace period is not specified, then all existing RPCs are 
+    teriminated immediately and the this method blocks until the last
+    RPC handler terminates.
 
-    This method does not block for any significant length of time. If None is
-    passed as the grace value, existing RPCs are immediately aborted and this
-    method blocks until this Server is completely stopped.
+    This method is idempotent, i.e. can be called multiple times. Passing a smaller
```

Comment:
Drop the ", i.e. can be called multiple times.". If the reader doesn't know what idempotence is, this isn't the place to teach them.